src/game_api/level_api.hpp

```diff
@@ -499,6 +508,8 @@ struct LevelGenSystem
 
     uint32_t get_procedural_spawn_chance(uint32_t chance_id);
     bool set_procedural_spawn_chance(uint32_t chance_id, uint32_t inverse_chance);
+
+    ~LevelGenSystem(){}; // cuz it was complaining
```

Comment:
Where was it complaining? Are you invoking this function somewhere? Otherwise it would not complain, that needs to be fixed.

Reply:
When i added the `union` compiler started throwing `warning C4624: 'LevelGenSystem': destructor was implicitly defined as deleted`, i don't get why in this scenario it cares about the deconstructor, unless there is some code that calls it

Reply:
It does sound like there is code that calls it, otherwise it should complain. If you look for the error in the Output window there should be more info.

Reply:
There is nothing more the in the output, just the standard `the following warning is treated as an error` and then the warning that i posted on `level_api.hpp(511)`

Before that it builds the render_api which has only one line ralated to the level_api and it's `texture_id = State::get().ptr_local()->current_theme->get_dynamic_texture(texture_id);`